components/style/properties/longhand/box.mako.rs

```diff
@@ -1196,14 +1194,37 @@ ${helpers.predefined_type("scroll-snap-coordinate",
                 m32: Number::from_computed_value(&computed.m32),
                 m33: Number::from_computed_value(&computed.m33),
                 m34: Number::from_computed_value(&computed.m34),
-                m41: Number::from_computed_value(&computed.m41),
-                m42: Number::from_computed_value(&computed.m42),
-                m43: Number::from_computed_value(&computed.m43),
+                m41: Either::Second(Number::from_computed_value(&computed.m41)),
+                m42: Either::Second(Number::from_computed_value(&computed.m42)),
+                m43: Either::Second(Number::from_computed_value(&computed.m43)),
                 m44: Number::from_computed_value(&computed.m44),
             }
         }
     }
 
+    /// Takes a computed LengthOrPercentageOrNumber and returns computed Number as CSSFloat.
+    /// This function is needed for -moz-transform because prefixed property treats length and
+    /// percentages as Number values.
+    ///
+    /// We needed to divide length to 60 to get actual pixel value.
+    /// We could also pass specified LengthOrPercentageOrNumber to be able to prevent this division
+    /// but Length and Calc have so many values to be calculated. Probably it's better to
+    /// leave these calculations to ToComputedValue method.
+    fn length_to_computed_number(value: computed::LengthOrPercentageOrNumber) -> CSSFloat {
+        match value {
+            Either::First(length) => {
+                match length {
+                    computed::LengthOrPercentage::Length(len) => (len.0 / 60) as CSSFloat,
+                    computed::LengthOrPercentage::Percentage(percentage) => percentage,
```

Comment:
It looks like gecko is convert to number while computing but doesn't serialize it.
See with and without percentage value in this example, it's same:
```html
<!DOCTYPE html>
<style>
div {
  width: 100px;
  height: 100px;
  background-color: red;
  -moz-transform: matrix(1, 0.3, 0, 1, 500%, 0);
}
</style>
<div></div>
```
Probably this is a gecko bug.